sympy/printing/fcode.py

```diff
@@ -106,6 +106,15 @@ def _get_statement(self, codestring):
 
     def _get_comment(self, text):
         return "! {0}".format(text)
+    
+    #issue 12267
+    def _print_sign(self,func):
+        if func.args[0].is_integer:
+            return "merge(0, isign(1, {0}), {0} == 0)".format(self._print(func.args[0]))
+        elif func.args[0].is_complex:
+            return "merge(cmplx(0d0, 0d0), {0}/abs({0}), abs({0}) == 0d0)".format(self._print(func.args[0]))
```

Comment:
Sorry, for me changing my mind. But @Franjov pointed out that we are already using the lower precision form om `cmplx` in other places in `fcode`. So you can disregard that comment. If tests passes now this is good to merge without any further changes.